Services/Rules.cs

```
@@ -85,13 +86,21 @@ public class Rules : IRules
 
         public async Task DeleteAsync(string id)
         {
-            if (Regex.IsMatch(id, INVALID_CHARACTER))
+            Rule existing = await this.GetAsync(id);
+            if (existing.Deleted)
             {
-                this.log.Debug("id contains illegal characters.", () => new { id });
-                throw new InvalidInputException("id contains illegal characters.");
+                throw new ResourceNotFoundException($"Rule {id} not found");
```

delete should fail silently when a resource doesn't exist or is already deleted - reporting OK

---

ping

---

already fixed, I think you are looking at outdated code